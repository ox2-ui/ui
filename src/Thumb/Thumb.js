import React, { PropTypes } from 'react';
import classNames from 'classnames';
import Icon from '@ox2/icon/Icon';
import ImageOffline from '@ox2/image/ImageOffline';

/**
 * Thumb Component
 */
const Thumb = (props) => {
  const { className: classNameProp, image, icon, width, height, offlineEnabled, localImg } = props;

  const className = classNames(
    'display:flex flex:items-center flex:content-center',
    classNameProp
  );

  return (
    <div className={className} style={{width: `${width}px`, height: `${height}px`, WebkitFlex: `0 0 ${width}px`, flex: `0 0 ${width}px`}}>
      {do {
        if (image) {
          if (offlineEnabled && localImg) {
            <ImageOffline remoteLink={image} localImg={localImg} className="w:100p border:rounded" />;
          } else {
            <img src={image} className="w:100p border:rounded" />;
          }
        } else {
          <Icon icon={icon} size={`${height - 20}px`} className="text-color:neutral-faded" />;
        }
      }}
    </div>
  );
};

Thumb.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Thumb height in pixels
   */
  height: PropTypes.number,
  /**
   * Icon id
   */
  icon: PropTypes.string,
  /**
   * Image url
   */
  image: PropTypes.string,
  /**
   * Offline/local server image filename
   */
  localImg: PropTypes.string,
  /**
   * If true will try to fetch local/offline images
   */
  offlineEnabled: PropTypes.bool,
  /**
   * Thumb width in pixels
   */
  width: PropTypes.number,
};

export default Thumb;
