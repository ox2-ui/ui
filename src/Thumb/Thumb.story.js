import React from 'react';
import { storiesOf } from '@storybook/react';
import Thumb from './Thumb';

const image1 =
  'https://randomuser.me/api/portraits/women/50.jpg';
const image2 =
  'https://randomuser.me/api/portraits/women/18.jpg';
const logo1 =
  'https://upload.wikimedia.org/wikipedia/commons/thumb/c/c6/Rohde_%26_Schwarz_Logo.svg/800px-Rohde_%26_Schwarz_Logo.svg.png';

const localImg = 'ktLb8Hqxo28WdToq5.jpg';

storiesOf('Thumb', module)
  .add('square image', () =>
    <Thumb image={image1} width={60} height={60} />,
  )
  .add('offline square image', () =>
    <Thumb
      image={image1}
      offlineEnabled={true}
      localImg={localImg}
      width={60}
      height={60}
    />,
  )
  .add('landscape logo', () =>
    <Thumb image={logo1} width={120} height={60} />,
  )
  .add('square white bg image', () =>
    <Thumb image={image2} width={60} height={60} />,
  )
  .add('icon placeholder', () =>
    <Thumb
      icon="user"
      iconGroup="linear"
      width={60}
      height={60}
    />,
  )
  .add('icon placeholder colored', () =>
    <Thumb
      icon="user"
      iconGroup="linear"
      width={60}
      height={60}
      iconColor={'brand'}
    />,
  )
  .add('icon placeholder dark bg', () =>
    <Thumb
      icon="user"
      iconGroup="linear"
      width={60}
      height={60}
      color={'neutral'}
      iconColor={'white'}
    />,
  )
  .add('icon placeholder small', () =>
    <Thumb
      icon="user"
      iconGroup="linear"
      width={30}
      height={30}
    />,
  )
  .add('square image 80', () =>
    <Thumb image={image1} width={80} height={80} />,
  )
  .add('square 80 icon placeholder', () =>
    <Thumb
      icon="user"
      iconGroup="linear"
      width={80}
      height={80}
    />,
  );
