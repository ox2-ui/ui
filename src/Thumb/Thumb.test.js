/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Thumb from './Thumb';

const image1 = 'https://randomuser.me/api/portraits/women/50.jpg';
const logo1 = 'https://upload.wikimedia.org/wikipedia/commons/thumb/c/c6/Rohde_%26_Schwarz_Logo.svg/800px-Rohde_%26_Schwarz_Logo.svg.png';


it('renders correctly as square', () => {
  const wrapper = mount(
    <Thumb image={image1} width={60} height={60} />
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly as landscape', () => {
  const wrapper = mount(
    <Thumb image={logo1} width={120} height={60} />
  );

  expect(wrapper).toMatchSnapshot();
});
