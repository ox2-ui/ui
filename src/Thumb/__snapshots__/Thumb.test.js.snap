exports[`test renders correctly as landscape 1`] = `
<Thumb
  height={60}
  image="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c6/Rohde_%26_Schwarz_Logo.svg/800px-Rohde_%26_Schwarz_Logo.svg.png"
  width={120}>
  <div
    className="display:flex flex:items-center flex:content-center border:rounded color:neutral"
    style={
      Object {
        "WebkitFlex": "0 0 120px",
        "flex": "0 0 120px",
        "height": "60px",
        "width": "120px",
      }
    }>
    <img
      className="w:100p border:rounded"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c6/Rohde_%26_Schwarz_Logo.svg/800px-Rohde_%26_Schwarz_Logo.svg.png" />
  </div>
</Thumb>
`;

exports[`test renders correctly as square 1`] = `
<Thumb
  height={60}
  image="https://randomuser.me/api/portraits/women/50.jpg"
  width={60}>
  <div
    className="display:flex flex:items-center flex:content-center border:rounded color:neutral"
    style={
      Object {
        "WebkitFlex": "0 0 60px",
        "flex": "0 0 60px",
        "height": "60px",
        "width": "60px",
      }
    }>
    <img
      className="w:100p border:rounded"
      src="https://randomuser.me/api/portraits/women/50.jpg" />
  </div>
</Thumb>
`;
